### PR TITLE
Generic repository pattern and unit of work added

### LINKED ISSUE #46: Repository implementation could be smarter
Link:
https://docs.microsoft.com/en-us/aspnet/mvc/overview/older-versions/getting-started-with-ef-5-using-mvc-4/implementing-the-repository-and-unit-of-work-patterns-in-an-asp-net-mvc-application

https://garywoodfine.com/generic-repository-pattern-net-core/

### PR DESCRIPTION
**Pull request by MadsDueKristensen**
Resolve #46

### CODE REVIEW

**Comment by MadsDueKristensen** [comment]
Many of the lines are just moved code

File: TaskHouseApi/Persistence/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            return true;
+        }
+
+        public bool isInDatabase(int Id)
+        {
+            if (!postgresContext.Users.Any(o => o.Id == Id))
+            {
+                return false;
+            }
+            return true;
```
Comment (kvartborg) [inline TaskHouseApi/Persistence/Repositories/UserRepository.cs]: No need to have an if statement.
```cs
return postgresContext.Users.Any(o => o.Id == Id)
```

File: TaskHouseApi/Persistence/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+         */
+        public bool DeleteRefrechToken(RefreshToken refreshToken)
+        {
+            postgresContext.Remove(refreshToken);
+            int affected = postgresContext.SaveChanges();
+            if (affected != 1)
+            {
+                return false;
+            }
+            return true;
```
Comment (kvartborg) [inline TaskHouseApi/Persistence/Repositories/UserRepository.cs]:
```cs
return affected == 1
```

File: TaskHouseUnitTests/FakeUserRepository.cs [hunk 3]
```diff
@@ -1,107 +1,88 @@
...
+        //         new Employer()
+        //         {
+        //             Id = 1,
+        //             Username = "1234",
+        //             Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+        //             Email = "test@test.com",
+        //             FirstName = "Bob1",
+        //             LastName = "Bobsen1",
+        //             Salt = "upYKQSsrlub5JAID61/6pA==",
+        //             Discriminator = "Employer"
```
Comment (kvartborg) [inline TaskHouseUnitTests/FakeUserRepository.cs]: This should properly be deleted?

File: TaskHouseUnitTests/FakeWorkerRepository.cs [hunk 4]
```diff
@@ -1,19 +1,59 @@
...
+                    Salt = "U+cUJhQU56X+OCiGF9hb1g==",
+                    Discriminator = "Worker"
+                }
+       }
+       )
+        { }
+        // public override IEnumerable<User> RetrieveAll()
+        // {
+        //     return list.Where(w => w.Discriminator == nameof(Worker));
+        // }
```
Comment (kvartborg) [inline TaskHouseUnitTests/FakeWorkerRepository.cs]: Remove or use?

File: TaskHouseApi/Persistence/DatabaseContext/PostgresContext.cs [hunk 5]
```diff
@@ -29,7 +29,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<User>().ToTable("Users");
 
 
-            Seed.SeedData(modelBuilder);
+            //Seed.SeedData(modelBuilder);
```
Comment (grammeaway) [inline TaskHouseApi/Persistence/DatabaseContext/PostgresContext.cs]: Needed? Else, remove

**Pull request merged**